api-flask/app/zonal_stats.py

```diff
@@ -193,14 +194,27 @@ def calculate_stats(
         stats_input = [s.get('geom') for s in zones]
         prefix = None
 
+    # Add function to calculate overlap percentage.
+    add_stats = None
+    if intersect_threshold:
+        def percentage_over_threshold(masked):
```

Comment:
IMO, we could refactor the function like this.
```python
        def percentage_over_threshold(masked):
            # Get total number of elements in the array.
            total = masked.size
            if total == 0:  # Avoid division by zero.
                return total

            # Count number of elements higher than threshold.
            over_threshold = (masked > intersect_threshold).sum()

            # Compute percentage over total.
            return over_threshold / total
```

Reply:
I think size does not take into account the shape of the admin boundary and returns the size of the complete unmasked area, no?

Reply:
https://numpy.org/doc/stable/reference/generated/numpy.ma.MaskedArray.size.html

Reply:
@JorgeMartinezG please let me know what you think about the above.
And have a look at the new format which now accepts parameters of the form `>=10`